### ISSUE TITLE
importing disnake.ext.commands causes errors in 3.11.0

### ISSUE THREAD

**Issue by rf20008** [DisnakeDev/disnake#823]
### Summary

importing disnake.ext.commands causes errors in 3.11.0

### Reproduction Steps

I installed Python 3.11 and tried to import disnake.ext.commands but it didn't work

### Minimal Reproducible Code

```python
from disnake.ext import commands
```

### Expected Results

I expected the import to actually import fine.

### Actual Results

I obtained this error:
```py
Traceback (most recent call last):
  File "<pyshell#2>", line 1, in <module>
    from disnake.ext import commands
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/disnake/ext/commands/__init__.py", line 23, in <module>
    from .flag_converter import *
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/disnake/ext/commands/flag_converter.py", line 50, in <module>
    @dataclass
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 1221, in dataclass
    return wrap(cls)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 1211, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 959, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/dataclasses.py", line 816, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'disnake.utils._MissingSentinel'> for field name is not allowed: use default_factory```

### Intents

(this is an issue with imports)

### System Information

```markdown
- Python v3.11.0-final
- disnake v2.6.1-final
    - disnake pkg_resources: v2.6.1
- aiohttp v3.8.3
- system info: Darwin 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:17:10 PDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64
```

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

This may be a problem caused by the breaking changes of Python 3.11 (and it's probably not supported yet)

**Comment by EmreTech** [comment]
FYI, you didn't format the code block correctly.
Code blocks are supposed to look like this on GitHub:
\`\`\`
<insert stuff here>
\`\`\`

**Comment by onerandomusername** [comment]
This is fixed on the master branch, as part of #785
